onnx/defs/tensor/defs.cc

```diff
@@ -492,7 +492,33 @@ Takes a  parameter `axes` with a list of axes to squeeze.
     .TypeConstraint(
         "T",
         {"tensor(float16)", "tensor(float)", "tensor(double)"},
-        "Constrain input and output types to float tensors.");
+        "Constrain input and output types to float tensors.")
+    .ShapeInferenceFunction([](InferenceContext& ctx) {
+        if (!hasExactlyNInputTypes(ctx, 1, "Squeeze")) {
+          return;
+        }
+
+        propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+        std::vector<int64_t> axes;
+        if (!getRepeatedAttribute(ctx, "axes", axes)) {
+          return;
+        }
+
+        if (!ctx.getInputType(0)->has_shape()) {
+          return;
+        }
+
+        ctx.getOutputType(0)->mutable_shape();
+
+        for (int i = 0, j = 0; i < ctx.getInputType(0)->shape().dim_size(); ++i) {
+          if (j < axes.size() && axes[j] == i) {
```

Comment:
This requires `axes` to be sorted. (which is unspecified in the spec)

Reply:
++